src/scc.js

```diff
@@ -0,0 +1,91 @@
+import calculateScc from '@rtsao/scc';
+import { hashObject } from 'eslint-module-utils/hash';
+import resolve from 'eslint-module-utils/resolve';
+import ExportMapBuilder from './exportMap/builder';
+import childContext from './exportMap/childContext';
+
+let cache = new Map();
+let exportMapBuilder = ExportMapBuilder;
+
+export default class StronglyConnectedComponentsBuilder {
+  static injectExportMapBuilder(value) {
+    exportMapBuilder = value;
+  }
+
+  static resetExportMapBuilder() {
+    exportMapBuilder = ExportMapBuilder;
+  }
```

Comment:
is there a better way to do this than manual injection methods? even a public property mocked with sinon would be preferable.

Reply:
What exactly would you like made better?
How is overriding a public property with a mock better?
Dependency injection in the constructor is a well known design pattern, and it's good to not surprise the reader.

Reply:
It's not an idiomatic pattern in JS, though, and adding it's better because adding extra code just for tests is a huge antipattern.

Reply:
> adding extra code just for tests is a huge antipattern.

That's a surprising statement. I'd like to present an opposing view, where adding extra code just for tests is a good thing, explained well by Robert C Martin in his book Clean Architecture on pages 251-252:

> The extreme isolation of tests, combined with the fact that they are not usually deployed, often causes developers to think that tests fall outside of the design of the system. This is a catastrophic point of view. Tests that are not well integrated into the design of the system tend to be fragile, and they make the system rigid and difficult to change.
> The solution is to design for testability. Don't depend on volatile things.
> The way to accomplish this goal is to create a specific API that the tests can use [...]
> The purpose of the testing API is to decouple the tests from the application. The goal is to decouple the structure of the tests from the structure of the application.

Reply:
https://qntm.org/clean might be a helpful read.

Reply:
Clean Code is a different book.

Does the blog you linked to say anything about adding extra code just for tests?

Reply:
That's totally a fair point - in that both approaches are fully globally mutable - but as a property, there's no code that goes unexecuted in production, which is still better than the current situation.

A tight coupling imo is preferable to test-only production code (neither are ideal).

Reply:
Got rid of injection by stubbing the imported dependency using sinon:

- https://github.com/import-js/eslint-plugin-import/pull/2998/commits/f64a1232edb46f12725cc466b0e4eb4f1237ec7a